packages/server/src/utilities/rowProcessor/index.ts

```diff
@@ -99,10 +105,17 @@ export async function processAutoColumn(
           row[key] = schema.lastID + 1
           schema.lastID++
           table.schema[key] = schema
+          tableMutated = true
         }
         break
     }
   }
+
+  if (tableMutated) {
+    const db = context.getAppDB()
+    const resp = await db.put(table)
+    table._rev = resp.rev
```

Comment:
This is a substantial change that sort-of doesn't belong in this PR but made a few things in this PR much easier.

Up until now, when calling `inputProcessing`, there was a change that the `Table` you pass in gets modified. This happens if you're using an internal table, it has an auto ID column, and you're creating a row. Scattered throughout the codebase were a lot of places where we cloned the `Table` prior to calling `inputProcessing` and diff old and new afterwards, saving if they are different.

These checks became a pain when it was not `Table` anymore but `Table | ViewV2` so I decided enough of this, I'm just going to save the table at the point it changes. This takes the complexity away from higher up the stack at the code of possibly doing more saves than before.

Reply:
Think this is fine - the `auto ID` is used a lot more rarely now that it used to be - this also explains why the worry about cloning has disappeared!